marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs

```diff
@@ -299,17 +299,31 @@ function saveNewJob(job) {
           } else {
             hubUtils.hubTrace(consts.TRACE_FLOW, `Inserting job report with URI: ${reportUri}`);
             hubUtils.writeDocument(reportUri, jobReport, permissions, collections, config.JOBDATABASE);
-          }  
+          }
         }
       }
     }
   }
 
+  function findProvenanceRecordUriFromJobId(jobId, targetDatabase) {
```

Comment:
@ryanjdew  @rahulvudutala  A refactor to consider (not as part of this ticket) - this module should become "job-lib.sjs" in the data-hub/5/flow directory. We'd then have "job.sjs" as a class module for wrapping a job document, and then multiple library modules for various purposes (which job.sjs would delegate to as appropriate).